include/fmt/base.h

```diff
@@ -924,15 +925,23 @@ template <typename T> class buffer {
   }
 
   /// Appends data to the end of the buffer.
-  template <typename U> void append(const U* begin, const U* end) {
+  template <typename U>
+// Workaround for Visual Studio 2019 to fix error C2893: Failed to specialize
+// function template 'void fmt::v11::detail::buffer<T>::append(const U *,const
+// U *)'
+#ifndef FMT_DETAIL_EXPLICIT_INSTANTIATIONS
+  FMT_CONSTEXPR20
+#endif
```

Comment:
This looks weird and probably a (benign) ODR violation. Maybe we should disable `FMT_CONSTEXPR20` completely on this version of MSVC.

Reply:
Visual studio 2019 supports C++20 and I think disabling ``FMT_CONSTEXPR20`` for it is a bad idea.
I suggest disable ``FMT_CONSTEXPR20`` only for ``fmt::formatted_size`` (implemented in new commit).


But, I don't see any ODR violation here, since the function body is the same.
If ODR is violated after PR, it must be violated without it due to explicit instantiation

Reply:
I think the new version is cleaner, let's go with it.

Reply:
Another solution, but it will break library ABI: remove explicit instantiations for ``buffer<>::append``.

Version with disabled FMT_CONSTEXPR20 for ``fmt::formatted_size`` on Visual studio 2019 is ready for review.